### PR TITLE
feat(list): make list fields configurable

### PR DESCRIPTION
**Pull request by petrjasek**
SDNTB-527

### CODE REVIEW

**Review by nrvikas — COMMENTED** [review]
Hi @petrjasek , can we consider moving this configuration to the 'planning_types' collection where we have editor fields configuration already ? That way the planning config stays in one place.
Also it would be great to have ES6 React function syntax for new 'fields' components you've created.
Thanks.

**Comment by mdhaman** [comment]
@petrjasek agree with @nrvikas. lets keep config in one place.

**Comment by petrjasek** [comment]
hi, I don't think it's the same as planning types or profiles in superdesk, this should be more or less profile agnostic, defines what users need to see in lists. unless there is a usecase that it would show different fields for different types, which I'm not aware of

File: client/components/fields/description.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import PropTypes from 'prop-types';
+
+export function description({item}) {
```
Comment (petrjasek) [inline client/components/fields/description.jsx]: @nrvikas es6 react func you mean `export const description = ({item) => ...` ?

Comment (nrvikas) [reply client/components/fields/description.jsx]: Yes please.

**Comment by mdhaman** [comment]
We don't have content profiles like superdesk i.e. we are using `planning_types` as `UI config` for editor of each type. On that basis, if we have `list` config document in the `planning_types` collection so that all configs are in one place. I think that is what @nrvikas is suggesting as well. As a suggestion, we can store the UI config (list, advance search) in separate collection (not in planning types). 

We have ticket to configure "Advanced Search" panel for each view as well (SDESK-3490).  The ticket description is simple but I think nobody is using place field. Moreover, we have removed "slugline" field from event editor for our instance. Hence we cannot have the "slugline" field in advance search for events view. 

`Content profiles` like Superdesk needs more thought i.e. do we need multiple events/planning types per instance? How we pre-fill data between types for actions like `Create Planning item from event` and "Add as event" from planning item.

**Comment by petrjasek** [comment]
ok np, will give it a try

**Comment by petrjasek** [comment]
any feedback @mdhaman @nrvikas ? using planning_types now

**Comment by nrvikas** [comment]
Hi @petrjasek , it looks great.  We were thinking of refactoring this (by making 'list' as a separate document in planning_types collection) at a later stage when we include advanced search config as a separate document (as it cannot sit under just event and planning as we have a combined view). Would it be a concern for any client to reconfigure this at a later stage if 'list' is another document in planning_types with 'event', 'planning' as dict objects with its respective list configs ?

Just another thing I noted is that default fields are coming from server in case of Editor Config, but from front-end in the list config. Do you want to keep it this way ?

**Comment by petrjasek** [comment]
I don't think it's an issue, but also I'm not sure if it makes sense to be a separate document. search should work for both at the same time, but list item is always of a certain type.

I don't care really about the defaults, assuming it will be configured via planning_types so this is just a fallback, not a config really.

**Review by mdhaman — APPROVED** [review]

**Review by nrvikas — APPROVED** [review]

**Pull request merged**